### ISSUE TITLE
Request: reporting unnecessary `int`-cast after `math.ceil` and `math.floor`

### ISSUE THREAD

**Issue by jamesbraza** [pylint-dev/pylint#8693]
### Current problem

Please see this snippet:

```python
import math

assert isinstance(math.ceil(5.5), int)
assert isinstance(int(math.ceil(5.5)), int)  # Extra int cast
```

The `int` cast is unnecessary after `math.floor` and `math.ceil`.

### Desired solution

It would be cool if `pylint` was able to flag these as unnecessary `int`-casts.

**Comment by jamesbraza** [comment]
I guess this is sort of one of many cases where an `int`-cast is unnecessary, as some function internally convert the return to an `int`.

Perhaps there could be a new rule set oriented around unnecessary casts.  `int(1)` could be one of them.

**Comment by jacobtylerwalls** [comment]
`int(1)` is trivial to check for, and for calls, like math.ceil, but more importantly, anything, we can infer the call result, find out if `int` is the only possible type returned, and then raise the message. +1 from me (as long it's more general than math.ceil...)

**Comment by mbyrnepr2** [comment]
Let's also consider the case of #8863: _It'll be great to add check that will detect useless type conversions like float(1.0)._
